File: cpp/gtest/AfricanParrot.cpp
#include <stdexcept>
#include <algorithm>

#include "ParrotBase.h"
#include "AfricanParrot.h"
using namespace std;

AfricanParrot::AfricanParrot(int numberOfCoconuts)
: ParrotBase(), numberOfCoconuts(numberOfCoconuts) {}



double AfricanParrot::getSpeed() {
	return max(0.0, baseSpeed - loadFactor * numberOfCoconuts);
}

File: Python/african_parrot.py

from parrot import IParrot

LOAD_FACTOR = 9.0

class AfricanParrot(IParrot):
    
    def __init__(self, number_of_coconuts):
        self._number_of_coconuts = number_of_coconuts
    def speed(self):
        return max(0, self._base_speed() - self._load_factor() * self._number_of_coconuts)

    def _load_factor(self):
        return LOAD_FACTOR